supervisord/tests/test_supervisord_integration.py

```diff
@@ -0,0 +1,81 @@
+# (C) Datadog, Inc. 2018
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+import os
+from time import sleep
+
+import pytest
+import xmlrpclib
+
+from .common import PROCESSES, STATUSES, SUPERVISORD_CONFIG, BAD_SUPERVISORD_CONFIG, supervisor_check
+from datadog_checks.checks.base import AgentCheck
+from datadog_checks.dev import docker_run, get_docker_hostname
+
+
+CHECK_NAME = 'supervisord'
+HOST = get_docker_hostname()
+PORT = 19001
+HERE = os.path.dirname(os.path.abspath(__file__))
+URL = "http://{}:{}".format(HOST, PORT)
+
+
+@pytest.fixture(scope='session', autouse=True)
+def spin_up_supervisord():
+    with docker_run(compose_file=os.path.join(HERE, 'compose', 'supervisord.yaml'), endpoints=URL):
+        server = xmlrpclib.Server('http://localhost:19001/RPC2')
+        server.supervisor.startAllProcesses()
+        yield
+
+
+# Supervisord should run 3 programs for 10, 20 and 30 seconds
+# respectively.
+# The following dictionnary shows the processes by state for each iteration.
+PROCESSES_BY_STATE_BY_ITERATION = map(lambda x: dict(up=PROCESSES[x:], down=PROCESSES[:x], unknown=[]), range(4))
+
+
+def test_check(aggregator):
```

Comment:
should we run tests in two separate tox env? The name of the test modules (unit/integration) seems to suggest a clear separation